### PR TITLE
[INLONG-2376] add sdk-common module for supporting PB compression protocol format based on Protobuffer protocol.

### LINKED ISSUE #2376: [Feature] add sdk-common module for supporting PB compression protocol format based on Protobuffer protocol.
### Description

add dataproxy-common module for supporting new protocol format based on Protobuffer protocol.
Parent issue: #2375 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Title Name: [INLONG-2376][Inlong-sdk] add sdk-common module for supporting PB compression protocol format based on Protobuffer protocol.

Fixes #2376 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+public class EventUtils {
+
+    /**
+     * encode
+     * 
+     * @param  inlongGroupId
+     * @param  inlongStreamId
+     * @param  compressedType
+     * @param  events
+     * @return                MessagePack
```
Comment (imvan) [inline inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java:48]: too much space

File: inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        return packBuilder.build();
+    }
+
+    /**
+     * decodeSdkPack
+     * 
+     * @param  packBytes
+     * @param  offset
+     * @param  length
+     * @return             List,ProxyEvent
```
Comment (wardlican) [inline inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java:107]: ditto

**Comment by imvan** [comment]
lgtm

File: inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java (hunk 1 above)
Comment (luchunliang) [reply inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java]: fixed

File: inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java (hunk 2 above)
Comment (luchunliang) [reply inlong-sdk/sdk-common/src/main/java/org/apache/inlong/sdk/commons/protocol/EventUtils.java]: fixed

**Review by wardlican — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by imvan — APPROVED** [review]

**Pull request merged**